src/snowflake/snowpark/_internal/code_generation.py

```diff
@@ -0,0 +1,594 @@
+    to_import_from_module: Dict[str, Set[Tuple[str, str]]] = defaultdict(set)
+    # classes that should be generated source code
+    classes_to_generate: List[type] = []
+
+    header_text = CODE_AS_COMMENT_HINT if code_as_comment else CODE_HEADER
+    classes_text = ""
+
+    # 1. find objects referenced by functions including classes, methods, modules, global variables
+    find_target_func_objects_references(
+        func, to_import, ref_objects, classes_to_generate
+    )
+
+    # 2. deal with the referenced objects by types
+    func_text, global_vars_text = resolve_target_func_referenced_objects_by_type(
+        func, to_import, to_import_from_module, ref_objects, code_as_comment
+    )
+
+    # 3. deal with the classes defined in the same module as func's
+    for cls in classes_to_generate:
+        classes_text = f"{classes_text}{textwrap.dedent(inspect.getsource(cls))}"
+
+    # 4. deal with imports and alias
+    imports_str = resolve_target_func_imports(to_import, to_import_from_module)
+
+    # concatenating all the referenced parts
+    source_code_without_target_func = f"{header_text}{imports_str}{global_vars_text}{classes_text}{func_text}".rstrip()
+
+    # 5. handle func, remove the udf annotation
+    complete_source_code, func_assignment = handle_target_func_self_source_code(
+        func, source_code_without_target_func, code_as_comment
+    )
+
+    # 6. handle function assignment
+    complete_source_code = f"""\
+{complete_source_code}
+{f"func = {func_assignment}"}\
+""".strip()
+
+    # 7. if code as comment is true, prefix each line with '#'
+    if code_as_comment:
+        complete_source_code = comment_source_code(complete_source_code)
+
+    return complete_source_code.strip()
+
+
+def is_lambda(func: FunctionType) -> bool:
+    """
+    Check whether the target function is a lambda function.
+    """
+    return func.__name__ == "<lambda>"
+
+
+def get_lambda_code_text(code_text: str) -> str:
+    """
+    Extract the lambda expression from code text.
+
+    Args:
+        The original code text containing the lambda expression.
+
+    Returns:
+        The string of the lambda expression.
+
+    """
+    # add a wrapper to handle the case that the line of lambda source code does not include caller
+    # such that ast could parse the expression tree:
+    #     session.udf.register(
+    #         lambda x, y: x + y, ...
+    #     )
+    try:
+        source_ast = ast.parse(code_text)
+    except SyntaxError as exc:
+        if "cannot assign to lambda" in str(exc):
+            # handle case like:
+            # session.udf.register(
+            #    lambda x, y: x + y, ...
+            # )
+            code_text = f"wrapper({code_text})"
+        elif "unmatched ')'" in str(exc):
+            # handle case like:
+            # session.udf.register(
+            #    lambda x, y: x + y, ...)
+            code_text = f"wrapper({code_text}"
```

Comment:
I'm wondering if we could add `code_text` to the cases as well. This makes it easier to understand these handlings :)

Reply:
you mean adding some test cases? yes, I added some

Reply:
Ah I just meant what the `code_text` looks like in the example showed in comment :)

Reply:
I've updated the comment to include `code_text`